### PR TITLE
updated prop value validation

### PR DESCRIPTION
**Pull request by vshand11**
1) updated prop value validation function.
2) updated docker files

### CODE REVIEW

File: src/metadata_validator.py [hunk 1]
```diff
@@ -350,23 +350,33 @@ def validate_prop_value(self, value, prop_def):
...
+                    except ValueError as e:
+                        continue
+                if val is None:
+                    errors.append(create_error("Not a valid date", f"The value, {value}, is not a valid date!"))
 
             elif type ==  "boolean":
-                try:
-                    val = bool(value)
-                except ValueError as e:
+                if not isinstance(value, bool) and value not in ["yes", "true", "no", "false"]:
```
Comment (knockknockyoo) [inline src/metadata_validator.py:371]: Is it intended? if value is not boolean, it wont store any error.

if not isinstance(value, bool) or (not isinstance(value, bool) and value not in...???

File: src/metadata_validator.py [hunk 2]
```diff
@@ -350,23 +350,33 @@ def validate_prop_value(self, value, prop_def):
...
                     errors.append(create_error("Not a boolean", f"The value, {value}, is not a boolean!"))
             
             elif type ==  "array":
-                try:
-                    val = list(value)
-                except ValueError as e:
-                    errors.append(create_error("Not a list", f"The value, {value}, is not a list!"))
+                arr = value.split("*")
```
Comment (knockknockyoo) [inline src/metadata_validator.py:375]: Just curious, should we check if this is empty array?

**Review by knockknockyoo — CHANGES_REQUESTED** [review]

File: src/metadata_validator.py (hunk 2 above)
Comment (vshand11) [reply src/metadata_validator.py:375]: Hi Young,  it has been checked if the value is empty before call prop value validation.
  line 225 - 227

File: src/metadata_validator.py (hunk 1 above)
Comment (vshand11) [reply src/metadata_validator.py:371]: Allowed not a bool, but must in ["yes", "true", "no", "false"]

File: src/metadata_validator.py [hunk 3]
```diff
@@ -350,23 +350,33 @@ def validate_prop_value(self, value, prop_def):
...
-            elif type ==  "datetime" or type ==  "date":
+            elif type ==  "datetime":
                 try:
-                    val = datetime.strptime(value, '%m/%d/%y %H:%M:%S')
+                    val = datetime.strptime(value, DATETIME_FORMAT)
                 except ValueError as e:
-                    errors.append(create_error("Not a date/datetime", f"The value, {value}, is neither a datetime nor date!"))
+                    errors.append(create_error("Not a valid datetime", f"The value, {value}, is not a valid datetime!"))
+
+            elif type ==  "date":
```
[the remark this answers is not in the document]
Comment (vshand11) [reply src/metadata_validator.py:359]: ok

File: src/metadata_validator.py [hunk 4]
```diff
@@ -383,9 +393,17 @@ def check_permissive(value, permissive_vals):
...
 def check_boundary(value, min, max):
     errors = []
-    if min and value < min:
-        errors.append(create_error("Less than minimum", f"The value is less than minimum, {value} < {min}!"))
-    if max and value > max:
-        errors.append(create_error("More than maximum", f"The value is more than maximum, {value} > {min}!"))
+    if min and min.get(VALUE_PROP):
+        val = min.get(VALUE_PROP)
+        exclusive = min.get(VALUE_EXCLUSIVE)
+        if (exclusive and value <= val) or (not exclusive and value < val):
```
[the remark this answers is not in the document]
Comment (vshand11) [reply src/metadata_validator.py:399]: In data model, the exclusive is set to either True or False without exception.
if exclusive is True, the value can't equal the min or max. so if equal, it is invalid.

File: src/metadata_validator.py [hunk 5]
```diff
@@ -350,23 +350,33 @@ def validate_prop_value(self, value, prop_def):
                 if len(errs) > 0:
                     errors.extend(errs)
 
-            elif type ==  "datetime" or type ==  "date":
+            elif type ==  "datetime":
```
[the remark this answers is not in the document]
Comment (vshand11) [reply src/metadata_validator.py:353]: type is not a reserved keyword in python.  So it's ok.

**Review by knockknockyoo — APPROVED** [review]

**Pull request merged**